client/lib/cgutil/cpuset_manager_v2.go

```diff
@@ -0,0 +1,332 @@
+//go:build linux
+
+package cgutil
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/nomad/helper"
+	"github.com/hashicorp/nomad/lib/cpuset"
+	"github.com/hashicorp/nomad/nomad/structs"
+	"github.com/opencontainers/runc/libcontainer/cgroups"
+	"github.com/opencontainers/runc/libcontainer/cgroups/fs2"
+	"github.com/opencontainers/runc/libcontainer/configs"
+)
+
+const (
+	// CreationPID is a special PID in libcontainer used to denote a cgroup
+	// should be created, but with no process added.
+	//
+	// https://github.com/opencontainers/runc/blob/v1.0.3/libcontainer/cgroups/utils.go#L372
+	CreationPID = -1
+
+	// DefaultCgroupParentV2 is the name of Nomad's default parent cgroup, under which
+	// all other cgroups are managed. This can be changed with client configuration
+	// in case for e.g. Nomad tasks should be further constrained by an externally
+	// configured systemd cgroup.
+	DefaultCgroupParentV2 = "nomad.slice"
+
+	// rootless is (for now) always false; Nomad clients require root, so we
+	// assume to not need to do the extra plumbing for rootless cgroups.
+	rootless = false
+)
+
+// nothing is used for treating a map like a set with no values
+type nothing struct{}
+
+// null represents nothing
+var null = nothing{}
```

Comment:
I like what you're doing here but calling it `null` means we do `c.sharing[id] = null` to add an ID to the sharing set, which reads almost the opposite of what you're going for. Maybe if it were `c.sharing[id] = ok` or something like that?

(What I wouldn't do for a real set type in the stdlib!)